pkg/operator/controller/externaldns/deployment_test.go

```diff
@@ -4983,6 +5020,260 @@ func TestSecurityContextChanged(t *testing.T) {
 	}
 }
 
+func TestDeploymentStrategyChanged(t *testing.T) {
+	var (
+		twentyFivePercent = intstr.FromString("25%")
+		twentyPercent     = intstr.FromString("20%")
+		one               = intstr.FromInt(1)
+	)
+
+	for _, tc := range []struct {
+		name    string
+		current *appsv1.Deployment
+		desired *appsv1.Deployment
+		updated *appsv1.Deployment
+		changed bool
+	}{
+		{
+			name:    "Recreate added for first time",
```

Comment:
For this case, shouldn't you check the `updated` is the same as `desired`?

Reply:
`updated` field was removed after the support for the desired rolling strategy was droppped.